libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CloudAdapter.cs

```diff
@@ -181,7 +198,10 @@ private async Task ConnectAsync(HttpRequest httpRequest, IBot bot, CancellationT
                     {
                         // Start receiving activities on the socket
                         // TODO: pass asp.net core lifetime for cancellation here.
+                        var connectionId = Guid.NewGuid();
+                        _streamingConnections.TryAdd(connectionId, streamingActivityProcessor);
```

Comment:
Why is this a random GUID instead of something already tied to the conversation (e.g. the conversation ID)? Is it to avoid removing the wrong `streamingActivityProcessor` from the dictionary _if_ there was a scenario where two `streamingActivityProcessors` were created for the same conversation?

Reply:
At the time of creating a streaming connection, we do not have a conversation yet. So, I use a guid to keep track of the connection and remove the correct connection later using that guid.

Reply:
d'oh! Should have thought about that, thanks for the clarification!